server/migrations/usersMigration.js

```diff
@@ -12,17 +12,21 @@ async function migrateUsers() {
       .collection('users')
       .find({})
       .forEach((user) => {
-        const userPreferences = calculatePreferences(user)
+        const { siteSubjects, updatedPreferences } = calculatePreferences(user)
 
         appDatabase.collection('users').updateOne(
           { _id: ObjectId(user._id) },
           {
             $set: {
-              preferences: userPreferences,
+              preferences: updatedPreferences,
+              siteSubjects,
+              legacyPreferences: JSON.stringify(user.preferences),
```

Comment:
since user preferences will be changing shape, should we only set `legacyPreferences` the first time that this migration is run? Otherwise when we run it a second time we will lose the old preferences.

Reply:
https://github.com/AMP-SCZ/dpdash/pull/241/commits/b31bf8638cc0a6baac97cfab0627a18f21238b6b